### ISSUE TITLE
calabash 0.12.0: Unable to resolve dependencies: calabash-cucumber requires awesome_print (~> 1.2.0)

### ISSUE THREAD

**Issue by riteshdalal** [calabash/calabash-ios#668]
sudo gem install calabash-cucumber
ERROR:  While executing gem ... (Gem::DependencyError)
    Unable to resolve dependencies: calabash-cucumber requires awesome_print (~> 1.2.0)

If awesome_print 1.2.0 is installed manually then running calabash-ios gives the following error

calabash-ios
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:2007:in `raise_if_conflicts': Unable to activate run_loop-1.2.2, because awesome_print-1.2.0 conflicts with awesome_print (~> 1.6) (Gem::LoadError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1176:in `activate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1212:in `block in activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1198:in `each'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1198:in `activate_dependencies'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/specification.rb:1180:in `activate'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/rubygems/core_ext/kernel_gem.rb:48:in `gem'
	from /usr/bin/calabash-ios:22:in `<main>'

**Comment by krukow** [comment]
which version of Calabash are you using? 

cc: @jmoody please look at this

**Comment by promo888** [comment]
same issue.using calabash-cucumber (0.12.0) on mac

**Comment by riteshdalal** [comment]
@krukow same version as what @promo888 mentioned. I was trying to install calabash-cucumber version 0.12.0 on my macbook pro with OS X 10.10.2

**Comment by jmoody** [comment]
The problem is that calabash 0.12.0 requires run-loop >= 1.2.1.

Run-loop 1.2.2 requires awesome-print 1.6, but calabash 0.12.0 requires awesome-print 1.2.

I introduced this problem by releasing run-loop 1.2.2 which requires awesome-print 1.6 (which has an important fix).  I can't decide if this is a bug or simply a fact of life for a ruby developer

Before I explain how to work around this issue, I must mention that if you are using `sudo` to install gems, you should stop.  This is very dangerous and a very poor practice.  [1]

#### Use a Gemfile and bundler. [2][3]

This is the _de facto_ strategy for managing gem dependencies.  

Your Gemfile would look like this:

```
source 'https://rubygems.org'
gem 'calabash-cucumber', '0.12.0'
gem 'run_loop', '1.2.1'
```

You update and install your gems using:

```
$ bundle update
```

Whenever you execute a ruby command, you do in the context of `bundle exec`.

```
$ bundle exec cucumber
$ bundle exec calabash-ios sim reset
```

Often people use the Unix alias `be` for `bundle exec`.

#### Manually manage your gem dependencies.

If you are using `sudo`, it will be difficult for you to manage your gem dependencies (manually or with bundler).  You will probably need to delete some binaries from `/usr/bin` in addition to running the commands below.

```
$ gem update --system
$ gem uninstall -Vax --force --no-abort-on-dependent run_loop
$ gem uninstall -Vax --force --no-abort-on-dependent awesome_print
$ gem install run_loop --version 1.2.1
$ gem install calabash-cucumber --version 0.12.0
```

#### Links

- [1] https://github.com/calabash/calabash-ios/wiki/B2-Updating-your-run-loop-version#still-installing-gems-with-sudo
- [2] http://bundler.io/
- [3] https://github.com/calabash/calabash-ios/wiki/B2-Updating-your-run-loop-version

**Comment by promo888** [comment]
Awesome.After downgrade of run_loop its working
Thanks a lot

2015-01-14 10:38 GMT+02:00 Joshua Moody <notifications@github.com>:

**Comment by jmoody** [comment]
@riteshdalal Did these steps work for you?  If so, can you close this issue?  Thanks!

**Comment by promo888** [comment]
Works fine.
Thanks

2015-01-15 13:10 GMT+02:00 Joshua Moody <notifications@github.com>:

**Comment by jmoody** [comment]
This will be resolved by release 0.12.1.
